aptos-move/framework/aptos-framework/sources/account/account_abstraction.move

```diff
@@ -0,0 +1,203 @@
+module aptos_framework::account_abstraction {
+    use std::error;
+    use std::option::{Self, Option};
+    use std::signer;
+    use std::string::{Self, String};
+    use aptos_std::ordered_map::{Self, OrderedMap};
+    use aptos_framework::create_signer;
+    use aptos_framework::event;
+    use aptos_framework::function_info::{Self, FunctionInfo};
+    use aptos_framework::object;
+    use aptos_framework::auth_data::AbstractionAuthData;
+    use aptos_framework::permissioned_signer::is_permissioned_signer;
+    #[test_only]
+    use aptos_framework::account::create_account_for_test;
+
+    friend aptos_framework::transaction_validation;
+    #[test_only]
+    friend aptos_framework::account_abstraction_tests;
+
+    const EDISPATCHABLE_AUTHENTICATOR_IS_NOT_USED: u64 = 1;
+    const EFUNCTION_INFO_EXISTENCE: u64 = 2;
+    const EAUTH_FUNCTION_SIGNATURE_MISMATCH: u64 = 3;
+    const ENOT_MASTER_SIGNER: u64 = 4;
+
+    const MAX_U64: u128 = 18446744073709551615;
+
+    #[event]
+    struct UpdateDispatchableAuthenticator has store, drop {
+        account: address,
+        update: vector<u8>,
+        auth_function: FunctionInfo,
+    }
+
+    #[event]
+    struct RemoveDispatchableAuthenticator has store, drop {
+        account: address,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// The dispatchable authenticator that defines how to authenticates this account in the specified module.
+    /// An integral part of Account Abstraction.
+    enum DispatchableAuthenticator has key, copy, drop {
+        V1 { auth_functions: OrderedMap<FunctionInfo, bool> }
```

Comment:
how many different authenticators can be registered? 
is this guaranteed to never exceed a single storage slot? if not, maybe we should use BigOrderedMap?

on the other hand , limiting that to the single storage slot size might be a fine restriction

Reply:
that's something I am not sure. I cannot imagine an account needs so many slots. since this is an enum we can do that as V2 if necessary?

Reply:
yeah, fine with that, that makes sense